stress/shell/src/main/java/alluxio/stress/cli/MultiOperationStressMasterBench.java

```diff
@@ -0,0 +1,479 @@
+      // When to start recording measurements
+      long recordMs = mContext.getStartMs() + FormatUtils.parseTimeSize(mParameters.mWarmup);
+      mResult.setRecordStartMs(recordMs);
+
+      boolean useStopCount = mParameters.mStopCount != MasterBenchParameters.STOP_COUNT_INVALID;
+
+      long bucketSize = (mContext.getEndMs() - recordMs) / StressConstants.MAX_TIME_COUNT;
+
+      long waitMs = mContext.getStartMs() - CommonUtils.getCurrentMs();
+      if (waitMs < 0) {
+        throw new IllegalStateException(String.format(
+            "Thread missed barrier. Increase the start delay. start: %d current: %d",
+            mContext.getStartMs(), CommonUtils.getCurrentMs()));
+      }
+      CommonUtils.sleepMs(waitMs);
+
+      long totalCounter;
+      int operationIndex = 0;
+      while (true) {
+        if (Thread.currentThread().isInterrupted()) {
+          break;
+        }
+        if (!useStopCount && CommonUtils.getCurrentMs() >= mContext.getEndMs()) {
+          break;
+        }
+        totalCounter = mContext.getTotalCounter().incrementAndGet() - 1;
+        if (useStopCount && totalCounter >= mParameters.mStopCount) {
+          break;
+        }
+
+        long operationCount;
+        operationIndex = operationIndexSelector.apply(operationIndex);
+        operationCount = mContext.getOperationCounter(operationIndex).incrementAndGet() - 1;
+        long startNs = System.nanoTime();
+        applyOperation(operationIndex, operationCount);
+        long endNs = System.nanoTime();
+
+        long currentMs = CommonUtils.getCurrentMs();
+        // Start recording after the warmup
+        if (currentMs > recordMs) {
+          mResult.incrementNumSuccess(operationIndex, 1);
+
+          // record response times
+          long responseTimeNs = endNs - startNs;
+          mResponseTimeNs[operationIndex].recordValue(responseTimeNs);
+
+          // track max response time
+          long[] maxResponseTimeNs = mResult.getStatistics(operationIndex).mMaxResponseTimeNs;
+          int bucket =
+              Math.min(maxResponseTimeNs.length - 1, (int) ((currentMs - recordMs) / bucketSize));
+          if (responseTimeNs > maxResponseTimeNs[bucket]) {
+            maxResponseTimeNs[bucket] = responseTimeNs;
+          }
+        }
+      }
+    }
+
+    //Each thread will randomly pick an operation to do by operations ratio.
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runOperationsRatioBased() throws Exception {
+      double ratioSum = 0;
+      double last = 0;
+      double[] ratioArray = new double[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mOperationsRatio[i];
+      }
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioArray[i] = last + mParameters.mOperationsRatio[i] / ratioSum;
+        last = ratioArray[i];
+      }
```

Comment:
add some comments on how the ratio limiting works. e.g. this is creating a probabilistic distribution function, and inside `runInternal` it's choosing an operation according to the distribution.

Reply:
added comments for the new created getProbabilityDistribution() method.